node/src/Pos/Block/Slog/Logic.hs

```diff
@@ -196,13 +197,17 @@ type MonadSlogApply ssc ctx m =
 -- ↑ I reduced duplication by introducing 'slogCommon', but it wants
 -- more and I don't.
 
+-- | Flag determining whether to call BListener callback.
+newtype ShouldCallBListener = ShouldCallBListener Bool
+
```

Comment:
Instead of wrapping a `Bool` inside a `newtype`, and then unwrapping it later to compare to `True` and `False`, I'd prefer something like `data ShouldCallBListener = CallBListener | DontCallBListener`, and then matching directly on those constructors.

Just a minor thing, though.

Reply:
I agree with Philip here, that would be even better 😉